pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1176,12 +1183,44 @@ func (c *Controller) getCurrentMaxStorageNodesPerZone(
 	return 0, fmt.Errorf("storage disabled")
 }
 
+func getDefaultStorageNodesDisaggregatedMode(nodeList *v1.NodeList, cluster *corev1.StorageCluster) (uint64, error) {
+	// Check if ENABLE_ASG_STORAGE_PARTITIONING is set to false. If not, we can look at the 'portworx.io/node-type' label
+	for _, envVar := range cluster.Spec.Env {
+		if envVar.Name == StoragePartitioningEnvKey {
+			if envVar.Value == "false" {
+				return 0, fmt.Errorf("%v is set to false", StoragePartitioningEnvKey)
+			}
+			break
+		}
+	}
+
+	// This will return a zone map of storage nodes in each zones
+	nodeTypeZoneMap := getZoneMap(nodeList, NodeTypeKey, StorageNodeValue)
+	// We'll find the zone with least number of labels
+	minValue := uint64(math.MaxUint64)
+	for _, value := range nodeTypeZoneMap {
+		if value < minValue {
+			minValue = value
+		}
+	}
+	return minValue, nil
+}
+
 // getDefaultMaxStorageNodesPerZone aims to return a good value for MaxStorageNodesPerZone with the
 // intention of having at least 3 nodes in the cluster.
-func getDefaultMaxStorageNodesPerZone(nodeList *v1.NodeList) (uint32, error) {
-	zoneMap := getZoneMap(nodeList)
+func getDefaultMaxStorageNodesPerZone(nodeList *v1.NodeList, cluster *corev1.StorageCluster) (uint32, error) {
+	var storageNodes uint64
+	var err error
+
+	// Check if storage nodes are explicitly set using labels
+	storageNodes, err = getDefaultStorageNodesDisaggregatedMode(nodeList, cluster)
+	if err == nil {
+		return uint32(storageNodes), nil
```

Comment:
if we set this as a fixed number, how can users scale storage nodes in the future?

Reply:
should we not set default maxStrorageNodesPerZone in disaggregated mode? as the user is managing it by the label anyway, feel like it's the same functionality but it adds more complexity to consider maxStrorageNodesPerZone while setting storage label.

Reply:
If we don't set it, we'll be in the same situation as before. Like I mentioned before, if a node pool has the storage label and it's upgraded, the new nodes that come in as replacement could possible form a new storage node unintentionally. 
I agree with you that it's an extra knob which affect UX. But I'd argue that without adding this we'll be adding more confusion.
With this change we can make a universal claim that if you expect to increase the storage nodes in a cluster the one thing you have to change is maxStroageNodesPerZone. And then they'll continue with the previous expectation that labels should be correct.
Without this change we will be putting an if else. If you have label and if dont 'have. have labels we have different behaviour.

Reply:
I'll include these scenarios in the documentation so that we avoid confusion and promote maxStorageNodesPerZone as the main knob.